### PR TITLE
Make BNCCorpusReader to parse the whole BNC

### PR DESCRIPTION
**Pull request by dimazest**
BNCCorpusReader doesn't handle all the tags available in the BNC. This patch add the missing tags.

In addition, the code is cleaned up and basic tests are written.

### CODE REVIEW

**Comment by stevenbird** [comment]
Thanks very much @dimazest - I'll merge this request. It would be helpful to know why the lazy loading code was removed.

File: nltk/corpus/reader/bnc.py [hunk 1]
```diff
@@ -100,16 +85,13 @@ def tagged_sents(self, fileids=None, c5=False, strip_space=True,
...
-                           for fileid in self.abspaths(fileids)])
-        else:
-            return concat([self._words(fileid, True, tag, strip_space, stem)
-                           for fileid in self.abspaths(fileids)])
+        tag = 'c5' if c5 else 'pos'
+        return self._views(fileids, sent=True, tag=tag, strip_space=strip_space, stem=stem)
+
+    def _views(self, fileids=None, sent=False, tag=False, strip_space=True, stem=False):
+        """A helper function that instantiates BNCWordViews or the list of words/sentences."""
+        f = BNCWordView if self._lazy else self._words
```
Comment (dimazest) [inline nltk/corpus/reader/bnc.py:93]: lazy loading is still here. it was the goal of the refactoring to get rid of duplicate code.

**Comment by ChrisHarvey-web** [comment]
Could you tell me How to turn BNC2014 into raw text?  My CorpusReader seems not work and I don't know why it gives me a empty list. code as follows:

from nltk.corpus.reader import BNCCorpusReader

corpora = BNCCorpusReader(root = 'C:/Users/86184/Desktop/BNC', fileids = '.*')

myfiles = corpora.words()

print(myfiles)

**Comment by stevenbird** [comment]
Please post on https://groups.google.com/forum/#!forum/nltk-users

**Comment by ChrisHarvey-web** [comment]
Hello Mr. What do you mean 'Whole BNC'? I have too many folds and files in BNC2014.&nbsp;






------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Steven Bird"<notifications@github.com&gt;;
发送时间:&nbsp;2020年6月5日(星期五) 上午7:31
收件人:&nbsp;"nltk/nltk"<nltk@noreply.github.com&gt;;
抄送:&nbsp;"青城小白龙"<437740611@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [nltk/nltk] Make BNCCorpusReader to parse the whole BNC (#654)





 
Please post on https://groups.google.com/forum/#!forum/nltk-users

**Pull request merged**